spec/waypoint_spec.js

```diff
@@ -872,26 +875,71 @@ describe('<Waypoint>', function() {
     });
   });
 
-  it('errors with a stateless component child', () => {
-    const errorMessage = 'You must wrap any Component Elements passed to Waypoint ' +
-      'in a DOM Element (eg; a <div>).';
-    const StatelessComponent = () => React.createElement('div');
-    this.props.children = React.createElement(StatelessComponent);
+  describe('when the Waypoint has children', () => {
+    it('does not throw with a DOM Element as a child', () => {
+      this.props.children = React.createElement('div');
+      expect(this.subject).not.toThrow();
+    });
+
+    it('errors when multiple children are provided', () => {
+      this.props.children = [
+        React.createElement('div'),
+        React.createElement('div'),
+        React.createElement('div')
+      ];
 
-    expect(this.subject).toThrowError(errorMessage);
-  });
+      expect(this.subject).toThrowError(notValidErrorMessage);
+    });
+
+    it('does not throw with a Statefull Component as a child', () => {
+      class StatefullComponent extends React.Component {
+        render() {
+          return React.createElement('div', { ref: this.props.innerRef });
+        }
+      }
+
+      this.props.children = React.createElement(StatefullComponent);
+      expect(this.subject).not.toThrow();
+    });
 
-  it('errors with a class-based component child', () => {
-    class ClassBasedComponent extends React.Component {
-      render() {
-        return React.createElement('div');
+    it('errors when a Statefull Component does not provide ref to Waypoint', () => {
+      class StatefullComponent extends React.Component {
+        render() {
+          return React.createElement('div');
```

Comment:
It's a little annoying that the test file isn't jsx (`<div/>` would have been simpler), but that's not your fault. :)

Reply:
I don't mind refactoring this if you want :) It seems that webpack config for karma allows to use JSX in `_spec` files

Reply:
That would be great! I'd prefer if that was a separate PR. And if you're making changes, can you rename the file and folder to be "test" instead of "spec"? We added this file right after being in Ruby land, so spec made sense. But I don't think javascript developers will look for "spec" files.

Reply:
Okay, I'll finish with this one and update tests!